### PR TITLE
Update LibHac to v0.16.0

### PR DESCRIPTION
**Pull request by Thealexbarney**
Main updates include:
- Support for reading NCAs with compressed sections. Fixes the following games:
  - Ryujinx/Ryujinx-Games-List#3815
  - Ryujinx/Ryujinx-Games-List#3786
  - Maybe others
- Be more resilient to invalid `DirectorySaveDataFileSystem` extra data states.
  Helps recover from situations where external things mess with the extra data files instead of just erroring
- Fix checking the FS access control content owner info size. 
  No actual games hit this issue

### CODE REVIEW

**Comment by GamerzHell9137** [comment]
iirc this is fixing the issue some people had with controllers that got rollbacked because the older version of libhac was bugged?

**Comment by gdkchan** [comment]
PR description should be edited, otherwise it will close the issues on the game list when merged.

**Comment by gdkchan** [comment]
Can confirm Gunvolt Chronicles: Luminous Avenger iX 2 works:
image
And Iridium:
image

File: Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/IApplicationFunctions.cs [hunk 1]
```diff
@@ -148,7 +148,7 @@ public ResultCode GetDesiredLanguage(ServiceCtx context)
             // TODO: When above calls are implemented, switch to using ns:am
 
             long desiredLanguageCode = context.Device.System.State.DesiredLanguageCode;
-            int  supportedLanguages  = (int)context.Device.Application.ControlData.Value.SupportedLanguages;
+            int  supportedLanguages  = (int)context.Device.Application.ControlData.Value.SupportedLanguageFlag;
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/IApplicationFunctions.cs:151]: I don't know what the official name is, but this name seems worse than the previous one, as it does not indicate there are "flags" for multiple languages there.

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Review by Thog — APPROVED** [review]

File: Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/IApplicationFunctions.cs (hunk 1 above)
Comment (Thealexbarney) [reply Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/IApplicationFunctions.cs:151]: I agree with you, although it does appear to be the official name https://switchbrew.org/wiki/NACP

**Pull request merged**